### PR TITLE
[Gamepad] Allow casting Town Portal, Teleport, and Guardian scrolls from inventory

### LINKED ISSUE #4273: [gamepad] Enable use of targeted spells from the inventory
A few spells can not be triggered from the inventory, there isn't any specific reason for this in most cases as they work without an  active target when using a gamepad.

https://github.com/diasurgical/devilutionX/pull/4266/files#diff-be3ecbe1ead4062ccd317dfeb0428b38adf3d54377059e602229c75bfc052da8R1813

### PR DESCRIPTION
**Pull request by StephenCWills**
This resolves #4273

### CODE REVIEW

File: Source/spells.cpp [hunk 1]
```diff
@@ -113,6 +113,16 @@ bool IsWallSpell(spell_id spl)
...
 }
 
+bool TargetsMonster(spell_id id)
+{
+	return id == SPL_FIREBALL
+	    || id == SPL_FIREWALL
+	    || id == SPL_FLAME
+	    || id == SPL_LIGHTNING
+	    || id == SPL_STONE
+	    || id == SPL_WAVE;
```
Comment (AJenbo) [inline Source/spells.cpp:123]: This is what took the longest to review, it's so confusing that fire `SPL_FIREBALL` targets monsters, but `SPL_FIREBOLT` does not :D

**Review by AJenbo — APPROVED** [review]
This will make support happy :smiling_face_with_tear:

File: Source/controls/plrctrls.cpp [hunk 2]
```diff
@@ -1861,39 +1861,52 @@ void PerformSpellAction()
...
-	if (pcursstashitem == uint16_t(-1))
+	if (pcursstashitem == uint16_t(-1)) {
 		return;
+	}
 
 	const Item &item = Stash.stashList[pcursstashitem];
-	if (item.IsScroll() && spelldata[item._iSpell].sTargeted) {
-		return;
+	if (item.IsScroll()) {
+		if (TargetsMonster(item._iSpell)) {
```
Comment (AJenbo) [inline Source/controls/plrctrls.cpp:1904]: Technically this won't happen as they are not allowed in town, but i like the consistency :+1:

File: Source/spells.cpp (hunk 1 above)
Comment (StephenCWills) [reply Source/spells.cpp:123]: Hmm, that's a good point. I hope this doesn't get used for anything other than scrolls...

**Pull request merged**